Library/Homebrew/cmd/vendor-install.sh

```diff
@@ -14,36 +14,37 @@ VENDOR_DIR="${HOMEBREW_LIBRARY}/Homebrew/vendor"
 
 # Built from https://github.com/Homebrew/homebrew-portable-ruby.
 set_ruby_variables() {
-  if [[ -n "${HOMEBREW_MACOS}" ]]
+  # Handle the case where /usr/local/bin/brew is run under arm64.
+  # It's a x86_64 installation there (we refuse to install arm64 binaries) so
+  # use a x86_64 Portable Ruby.
+  if [[ -n "${HOMEBREW_MACOS}" && "${VENDOR_PHYSICAL_PROCESSOR}" == "arm64" && "${HOMEBREW_PREFIX}" == "/usr/local" ]]
   then
-    if [[ "${VENDOR_PHYSICAL_PROCESSOR}" == "x86_64" ]] ||
-       # Handle the case where /usr/local/bin/brew is run under arm64.
-       # It's a x86_64 installation there (we refuse to install arm64 binaries) so
-       # use a x86_64 Portable Ruby.
-       [[ "${VENDOR_PHYSICAL_PROCESSOR}" == "arm64" && "${HOMEBREW_PREFIX}" == "/usr/local" ]]
+    ruby_PROCESSOR="x86_64"
+    ruby_OS="darwin"
+  else
+    ruby_PROCESSOR="${VENDOR_PHYSICAL_PROCESSOR}"
+    if [[ -n "${HOMEBREW_MACOS}" ]]
     then
-      ruby_FILENAME="portable-ruby-${HOMEBREW_PORTABLE_RUBY_VERSION}.el_capitan.bottle.tar.gz"
-      ruby_SHA="a5ef040e054444a0eb2cbcc1032fed14702dfbe2e55b25e609f3ce643f23c4ee"
-    elif [[ "${VENDOR_PHYSICAL_PROCESSOR}" == "arm64" ]]
+      ruby_OS="darwin"
+    elif [[ -n "${HOMEBREW_LINUX}" ]]
     then
-      ruby_FILENAME="portable-ruby-${HOMEBREW_PORTABLE_RUBY_VERSION}.arm64_big_sur.bottle.tar.gz"
-      ruby_SHA="49847c7a13f7094b211f6d0025900dd23716be07dac894a3d6941d7696296306"
+      ruby_OS="linux"
     fi
-  elif [[ -n "${HOMEBREW_LINUX}" ]]
+  fi
+
+  ruby_PLATFORMINFO="${HOMEBREW_LIBRARY}/Homebrew/utils/portable-ruby-sha256/${ruby_PROCESSOR}-${ruby_OS}"
+  if [[ -f "${ruby_PLATFORMINFO}" ]]
```

Comment:
What's the output look like if you run this on a platform where this file doesn't exist?

Reply:
```console
$ brew vendor-install --debug ruby
+ [[ -z '' ]]
+ VENDOR_PHYSICAL_PROCESSOR=arm64
+ [[ -z '' ]]
+ VENDOR_PROCESSOR=arm64
+ set_ruby_variables
+ [[ -n 1 ]]
+ [[ arm64 == \a\r\m\6\4 ]]
+ [[ /opt/homebrew == \/\u\s\r\/\l\o\c\a\l ]]
+ ruby_PROCESSOR=arm64
+ [[ -n 1 ]]
+ ruby_OS=darwin
+ ruby_PLATFORMINFO=/opt/homebrew/Library/Homebrew/vendor/portable-ruby-arm64-darwin
+ [[ -f /opt/homebrew/Library/Homebrew/vendor/portable-ruby-arm64-darwin ]]
+ [[ -n '' ]]
+ check_linux_glibc_version
+ [[ -z '' ]]
+ return 0
+ filename_var=ruby_FILENAME
+ sha_var=ruby_SHA
+ url_var=ruby_URL
+ VENDOR_FILENAME=
+ VENDOR_SHA=
+ VENDOR_URL=
++ cat /opt/homebrew/Library/Homebrew/vendor/portable-ruby-version
+ VENDOR_VERSION=3.3.3
+ [[ -z '' ]]
+ odie 'No Homebrew ruby 3.3.3 available for arm64 processors!'
+ onoe 'No Homebrew ruby 3.3.3 available for arm64 processors!'
+ [[ -n '' ]]
+ [[ -t 2 ]]
+ [[ -z '' ]]
+ echo -ne '\033[4;31mError\033[0m: '
Error: + [[ 1 -eq 0 ]]
+ echo 'No Homebrew ruby 3.3.3 available for arm64 processors!'
No Homebrew ruby 3.3.3 available for arm64 processors!
+ exit 1
```

Reply:
Great, thanks for checking!